crates/tauri-cli/src/build.rs

```diff
@@ -172,19 +172,36 @@ pub fn setup(
         ));
     }
 
+    // Issue #13287 - Allow the use of target dir inside frontendDist/distDir
+    // https://github.com/tauri-apps/tauri/issues/13287
+    let mut target_path = get_cargo_target_dir(&options.args)?;
+    target_path = tauri_path.join(target_path);
+    let target_path = fs::canonicalize(target_path)?;
+
     let mut out_folders = Vec::new();
-    for folder in &["node_modules", "src-tauri", "target"] {
-      if web_asset_path.join(folder).is_dir() {
+    if target_path.starts_with(web_asset_path) {
```

Comment:
web_asset_path is not canonicalized yet (the canonicalize call in line 169 does not carry over) so this check may be false for the same directories.